paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/StaticFileStoreSplitEnumerator.java

```diff
@@ -41,7 +42,8 @@
 
 /** A {@link SplitEnumerator} implementation for {@link StaticFileStoreSource} input. */
 public class StaticFileStoreSplitEnumerator
-        implements SplitEnumerator<FileStoreSourceSplit, PendingSplitsCheckpoint> {
+        implements SplitEnumerator<FileStoreSourceSplit, PendingSplitsCheckpoint>,
+                SupportsHandleExecutionAttemptSourceEvent {
```

Comment:
You may need to introduce classes for Flink 1.15.

Reply:
Could we remove 1.15 like remove spark-3.1(https://github.com/apache/paimon/issues/3941)?

We also has plan to support Flink-2.0, to avoid to maintain too many Flink versions, could we remove the support for Flink-1.15?

@JingsongLi

Reply:
No, it is good to keep Flink 1.15 support.

Reply:
OK